tools/exp/filter/filter.go

```diff
@@ -0,0 +1,147 @@
+package filter
+
+import (
+	"errors"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/mitchellh/copystructure"
+)
+
+// errUnexpected is the error that is returned when an unexpected error occurs.
+var errUnexpected = errors.New("unexpected filtering error")
+
+// supportedServiceTypes is a pseudo-constant map of supported service types.
+var supportedServiceTypes = func() map[string]struct{} {
+	return map[string]struct{}{
+		"alertmanager":                    {}, // As of v3.5.1 this is not supported in Aiven Terraform provider.
+		"cassandra":                       {},
+		"clickhouse":                      {},
+		"datadog":                         {},
+		"elasticsearch":                   {},
+		"external_aws_cloudwatch_logs":    {},
+		"external_aws_cloudwatch_metrics": {},
+		"external_elasticsearch_logs":     {},
+		"external_google_cloud_logging":   {},
+		"external_kafka":                  {},
+		"external_opensearch_logs":        {},
+		"external_postgresql":             {},
+		"flink":                           {},
+		"grafana":                         {},
+		"influxdb":                        {},
+		"jolokia":                         {},
+		"kafka":                           {},
+		"kafka_connect":                   {},
+		"kafka_mirrormaker":               {},
+		"m3aggregator":                    {},
+		"m3coordinator":                   {},
+		"m3db":                            {},
+		"mysql":                           {},
+		"opensearch":                      {},
+		"pg":                              {},
+		"prometheus":                      {},
+		"redis":                           {},
+		"rsyslog":                         {},
+		"signalfx":                        {},
+		"sw":                              {}, // As of v3.5.1 this is not supported in Aiven Terraform provider.
+	}
+}
+
+// ServiceTypes is a filter that filters out service types that are not supported.
+func ServiceTypes(f map[string]aiven.ServiceType) (map[string]aiven.ServiceType, error) {
+	cf, err := copystructure.Copy(f)
+	if err != nil {
+		return nil, err
+	}
+
+	acf, ok := cf.(map[string]aiven.ServiceType)
+	if !ok {
+		return nil, errUnexpected
+	}
+
+	for k := range acf {
+		if _, ok = supportedServiceTypes()[k]; !ok {
+			delete(acf, k)
+		}
+	}
+
+	return acf, nil
+}
+
+// IntegrationTypes is a filter that filters out integration types that are not supported.
+func IntegrationTypes(f []aiven.IntegrationType) ([]aiven.IntegrationType, error) {
```

Comment:
We don't have to filter this since Aiven internal services that we should filter out are not using service integrations by design; hence all that's coming from API is available for the end user.

Reply:
they, in fact, are, for some reason